opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -216,11 +217,23 @@ namespace Opm {
     template<typename TypeTag>
     std::vector< Well >
     BlackoilWellModel<TypeTag>::
-    getLocalNonshutWells(const int timeStepIdx, int& globalNumWells) const
+    getLocalNonshutWells(const int timeStepIdx, int& globalNumWells)
```

Comment:
Suggest to make this a separate function, and keep the old function unchanged (and still const).

Reply:
You have a conflict with #2936 over this, which will cause more wasted time. Conceptually, a getter function should remain const. The new function does two things, and the new addition has different semantics (mutating state). I think that is very good reason for splitting it.

Reply:
... and this somehow broke #2954. Seems like calling two functions (`getLocalNonshutWells` followed by `createLocalParallelWellInfo`) in `initFromRestartFile` is to complex for me. 

 Quite surprising that jenkins on this PR did not segfault as in the other PR.

Reply:
turns out that is no surprise as changes trickering it are still missing. puh, that was close.